ClientGenerator/src/googleapis/codegen/languages/csharp/default/templates/_method.tmpl

```diff
@@ -134,11 +134,19 @@ public override string RestPath
 public Google.Apis.Download.IMediaDownloader MediaDownloader { get; private set; }
 
 /// <summary>Synchronously download the media into the given stream.</summary>
+[System.Obsolete("This method hides errors; use " + nameof(Download2) + "(), which returns the final download status.")]
 public virtual void Download(System.IO.Stream stream)
 {
     MediaDownloader.Download(this.GenerateRequestUri(), stream);
 }
 
+/// <summary>Synchronously download the media into the given stream.</summary>
+/// <returns>The final status of the download; including whether the download succeeded or failed.</returns>
+public virtual Google.Apis.Download.IDownloadProgress Download2(System.IO.Stream stream)
```

Comment:
`DownloadWithStatus`? `DownloadWithFinalProgress`? Not keen on `Download2` as a name at the moment. Let's come up with some other options...